src/rules/orderedImportsRule.ts

```diff
@@ -0,0 +1,137 @@
+import * as Lint from "../lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "ordered-imports",
+        description: "Requires that import statements be alphabetized.",
+        descriptionDetails: Lint.Utils.dedent`
+            Enforce a consistent ordering for ES6 imports:
+            - Named imports must be alphabetized (i.e. "import {A, B, C} from "foo";")
+                - The exact ordering can be controled by the named-imports-order option.
+                - "longName as name" imports are ordered by "longName".
+            - Import sources must be alphabetized within groups, i.e.:
+                    import * as a from "a";
+                    import * as b from "b";
+            - groups of imports are delineated by blank lines. You can use these to group imports
+                however you like, e.g. by first- vs. third-party or thematically.`,
+        optionsDescription: Lint.Utils.dedent`
+            You may set the \`"named-imports-order"\` option to control the ordering of named
+            imports (the \`{A, B, C}\` in \'import {A, B, C} from "foo"\`.)
+
+            Possible values for \`"named-imports-order"\` are:
+
+            * \`"case-insensitive'\`: Correct order is \`{A, b, C}\`. (This is the default.)
+            * \`"lowercase-first"\`: Correct order is \`{b, A, C}\`.
+            * \`"lowercase-last"\`: Correct order is \`{A, C, b}\`.
+        `,
+        options: {
+            type: "object",
+            properties: {
+                "named-imports-order": {
+                    oneOf: ["case-insensitive", "lowercase-first", "lowercase-last"],
+                },
+            },
+            additionalProperties: false,
+        },
+        optionExamples: ["true", '[true, {"named-imports-order": "lowercase-first"}]'],
+        type: "style",
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static NAMED_IMPORTS_UNORDERED = "Named imports must be alphabetized.";
+    public static IMPORT_SOURCES_UNORDERED = "Import sources within a block must be alphabetized.";
```

Comment:
Since we use the word "group" in the documentation above, lets also use it here. I'm afraid "block" will imply the `{  }` type of block